services/image-service.ts

```diff
@@ -101,12 +101,14 @@ class ImageService {
                     throw new ZoneValidationError(404, ZoneError.CAT_NOT_FOUND);
                 }
             }
-            cat.images.filter((item: string) => item !== imageID);
+
+            cat.images = cat.images.filter((item: string) => item != imageID);
```

Comment:
The implementation of the method seems based on the premise that imageID is a string, please confirm that it is not an option to change the argument type instead.